### PR TITLE
Remove runtime flag envoy_reloadable_features_immediate_response_use_filter_mutation_rule

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
Risk Level: low
Testing: n/a
Docs Changes: n/a
Release Notes: inline
Fixes: https://github.com/envoyproxy/envoy/issues/31977

Description:

https://github.com/envoyproxy/envoy/pull/27925 (Applying same ext_proc filter checker rule for immediate response.) introduced a feature guarded by: envoy_reloadable_features_immediate_response_use_filter_mutation_rule. It has been 6 months since the code has been exercised by default, so it's time to remove the old code path and deprecate this runtime flag.

### CODE REVIEW

File: source/extensions/filters/http/ext_proc/ext_proc.cc [hunk 1]
```diff
@@ -1249,16 +1249,9 @@ void Filter::sendImmediateResponse(const ImmediateResponse& response) {
...
-              "envoy.reloadable_features.immediate_response_use_filter_mutation_rule")) {
-        mut_status = MutationUtils::applyHeaderMutations(response.headers(), headers, false,
-                                                         config().mutationChecker(),
-                                                         stats_.rejected_header_mutations_);
-      } else {
-        mut_status = MutationUtils::applyHeaderMutations(
-            response.headers(), headers, false, config().immediateMutationChecker().checker(),
-            stats_.rejected_header_mutations_);
-      }
+      mut_status = MutationUtils::applyHeaderMutations(response.headers(), headers, false,
```
Comment (adisuissa) [inline source/extensions/filters/http/ext_proc/ext_proc.cc:1252]: nit: combine with previous line, and convert to `const`.

**Review by adisuissa — COMMENTED** [review]
Thanks!
Left a minor comment.

**Review by alyssawilk — COMMENTED** [review]
mind updating the PR description?
Also you want "Fixes issue" as that auto-closes the issue on merge

File: source/extensions/filters/http/ext_proc/ext_proc.cc (hunk 1 above)
Comment (yanjunxiang-google) [reply source/extensions/filters/http/ext_proc/ext_proc.cc]: done

**Comment by yanjunxiang-google** [comment]
> mind updating the PR description? Also you want "Fixes issue" as that auto-closes the issue on merge

Sure, done

**Comment by alyssawilk** [comment]
alas needs main merge now
/wait

**Comment by alyssawilk** [comment]
also for PR description please fill in all valid fields, and remove any fields which are unused

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**